recurring_contract/models/recurring_contract.py

```diff
@@ -303,28 +303,20 @@ def rewind_next_invoice_date(self):
                     contract.invoice_line_ids.filter_for_contract_rewind("not_paid")
                     .mapped("move_id.invoice_date") or [False])
 
-                rewind_invoice_date = latest_paid_invoice_date + \
-                                      contract.group_id.get_relative_delta() \
-                    if latest_paid_invoice_date else earliest_open_invoice_date
-
-                if rewind_invoice_date:
-                    res |= contract._clean_invoices(rewind_invoice_date)
-                    contract.with_context(no_clean_on_write=True).write({
-                        "next_invoice_date": rewind_invoice_date
-                    })
-                else:
+                rewind_invoice_date = earliest_open_invoice_date
+                if not rewind_invoice_date:
```

Comment:
You completely removed the use of `latest_paid_invoice_date` which I don't think is correct. I would reintroduce it and use it first to try to determine the `rewind_invoice_date`.
```suggestion
                if latest_paid_invoice_date:
                    rewind_invoice_date = latest_paid_invoice_date + contract.group_id.get_relative_delta()
                else:
                    rewind_invoice_date = earliest_open_invoice_date
                if not rewind_invoice_date:
```